.github/workflows/scheduled.yml

```diff
@@ -71,46 +70,161 @@ env:
 jobs:
   compile:
     name: Build
-    runs-on: 8-core
+    runs-on: 16-core
+    container: ghcr.io/facebookincubator/velox-dev:centos8
     timeout-minutes: 120
     env:
-      CCACHE_DIR: "${{ github.workspace }}/.ccache/"
-      CCACHE_BASEDIR: "${{ github.workspace }}"
+      CCACHE_DIR: "/__w/velox/velox/.ccache/"
       LINUX_DISTRO: "ubuntu"
+      MAKEFLAGS: "NUM_THREADS=${{ inputs.numThreads || 16 }} MAX_HIGH_MEM_JOBS=${{ inputs.maxHighMemJobs || 8 }} MAX_LINK_JOBS=${{ inputs.maxLinkJobs || 4 }}"
+
+    defaults:
+      run:
+        shell: bash
+        working-directory: velox
+    outputs:
+      bias_presto: ${{ steps.sig-check.outputs.presto }}
+      bias_spark: ${{ steps.sig-check.outputs.spark }}
+
     steps:
 
-      - name: "Restore ccache"
+      - name: Get Function Signature Stash
+        uses: assignUser/stash/restore@v1
+        id: get-sig
+        with:
+          path: /tmp/signatures
+          key: function-signatures
+
+      - name: Restore ccache
         uses: assignUser/stash/restore@v1
         with:
           path: "${{ env.CCACHE_DIR }}"
-          key: ccache-fuzzer
+          key: ccache-fuzzer-centos
+
+      - name: Fix git permissions
+        working-directory: ${{ github.workspace }}
+          # Usually actions/checkout does this but as we run in a container
+          # it doesn't work
+        run: |
+          git config --global --add safe.directory /__w/velox/velox/velox
+          git config --global --add safe.directory /__w/velox/velox/velox_main
+
+      - name: Ensure Stash Dirs Exists
+        working-directory: ${{ github.workspace }}
+        run: |
+          mkdir -p '${{ env.CCACHE_DIR }}'
+          mkdir -p /tmp/signatures
+          # this can be removed after #8917
+          dnf install -q -y python39-devel
+
+      - name: Checkout Main
```

Comment:
We should check out the merge base instead of main.

Reply:
Why? With the way `actions/checkout` works we are comparing current  HEAD of main with the merge commit. This means we don't get false hits on changes that are new on main but unchanged in the PR (because they are in the  merge commit) but we also don't hide any incompatibilities introduced via new changes to `main` since the PR was opened which would only show up after merge of the PR when using the merge base.

It would probably make sense to checkout `main~1` when running the fuzzer on push to main so we don't miss any new changes (otherwise it would just compare HEAD to HEAD which ofc wouldn't have changes).